zerver/migrations/0064_add_file_name_field_to_realm_emoji.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.5 on 2017-03-09 05:23
+from __future__ import unicode_literals
+
+import os
+import io
+import requests
+
+from mimetypes import guess_type
+
+from PIL import Image
+from PIL import ImageOps
+from django.db import migrations, models
+
+from django.db import migrations
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.conf import settings
+from boto.s3.key import Key
+from boto.s3.connection import S3Connection
+from requests import ConnectionError, Response
+from typing import Dict, Text, Tuple, Optional, Union
+
+from six import binary_type
+
+
+def force_str(s, encoding='utf-8'):
+    # type: (Union[Text, binary_type], Text) -> str
+    """converts a string to a native string"""
+    if isinstance(s, str):
+        return s
+    elif isinstance(s, Text):
+        return s.encode(str(encoding))
+    elif isinstance(s, binary_type):
+        return s.decode(encoding)
+    else:
+        raise TypeError("force_str expects a string type")
+
+
+class Uploader(object):
+    def __init__(self):
+        # type: () -> None
+        self.path_template = "{realm_id}/emoji/{emoji_file_name}"
+        self.emoji_size = (64, 64)
+
+    def upload_files(self, response, resized_image, dst_path_id):
+        # type: (Response, binary_type, Text) -> None
+        raise NotImplementedError()
+
+    def get_dst_path_id(self, realm_id, url, emoji_name):
+        # type: (int, Text, Text) -> Tuple[Text,Text]
+        _, image_ext = os.path.splitext(url)
```

Comment:
Can you please confirm whether the url here would always be having the image extension at the end. It feels like there has been no mechanism earlier to ensure that image url's have an extension at the end(I guess that was not needed earlier).